### PR TITLE
Adding deep-clone as js coding question

### PR DESCRIPTION
**Pull request by gauravbehere**
# Pull Request

Adding deep-clone as JS coding question

Fixes # (issue)
None

## Type of change

Please delete options that are not relevant.

- [x] New Question
- [ ] Revision of an existing question
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate) 

# Checklist:

- [x] My content follows the style guidelines of this project
- [x] I have performed a self-review of my own content

Pull requests should be thought of as a conversation. There will be some back and forth when trying to get code merged into this or any other project. With all but the simplest changes you can and should expect that the maintainers of the project will request changes to your code. Please be aware of that and check in after you open your PR in order to get your code merged in cleanly. 

Thanks!

### CODE REVIEW

File: src/questions/javascript-questions.md [hunk 1]
```diff
@@ -64,3 +64,4 @@ console.log("hello" || "world")
 console.log("foo" && "bar")
 ```
 * Write an immediately invoked function expression (IIFE)
+* Write a method to deep clone an object of any depth
```
Comment (roblarsen) [inline src/questions/javascript-questions.md:67]: I think a two parter might be good here. Defining deep vs. shallow. 

```suggestion
* What is the difference between a deep copy and a shallow copy of a JavaScript object. Can you write a method to deep clone an object of any depth?
```

File: src/questions/javascript-questions.md [hunk 2]
```diff
@@ -64,3 +64,5 @@ console.log("hello" || "world")
 console.log("foo" && "bar")
 ```
 * Write an immediately invoked function expression (IIFE)
+* What is the difference between a deep copy and a shallow copy of a JavaScript object?
```
Comment (vvscode) [inline src/questions/javascript-questions.md:67]: I personally against the questions about `knowing terms`, except the case it's part of the language

what you're going to do if on an interview I would say `I've never heard such words` ?

**Comment by gauravbehere** [comment]
With a lot of PERSONAL opinions & difference of opinions among members. I would prefer to add my own list rather.
Why dont we have a grammar guideline as well ;)
Closing this PR. :)

File: src/questions/javascript-questions.md (hunk 2 above)
Comment (roblarsen) [reply src/questions/javascript-questions.md:67]: I often interview people who don't know certain terms. I continue the conversation either skipping over to the next question or I define what I'm asking about and then move onto the second part. Sometimes, people know the topic but just don't know the technical term.

**Comment by roblarsen** [comment]
@gauravbehere Sorry if you're frustrated, but disagreement is a healthy part of the open source process and back and forth is how disagreement is worked out. I talk about it specifically: 

>Pull requests should be thought of as a conversation. There will be some back and forth when trying to get code merged into this or any other project. With all but the simplest changes you can and should expect that the maintainers of the project will request changes to your code. Please be aware of that and check in after you open your PR in order to get your code merged in cleanly.

**Pull request closed without merging**